lib/mutant/matcher/compiler.rb

```diff
@@ -34,60 +34,23 @@ def call(subject)
 
     private
 
-      # Predicate constraining matches
+      # Predicate returning false on ignored subject
       #
       # @return [#call]
       #
-      # rubocop:disable MethodLength
-      #
       # @api private
-      def predicate
-        if subject_selector && subject_rejector
-          Morpher::Evaluator::Predicate::Boolean::And.new([
-            subject_selector,
-            Morpher::Evaluator::Predicate::Negation.new(subject_rejector)
-          ])
-        elsif subject_selector
-          subject_selector
-        elsif subject_rejector
-          Morpher::Evaluator::Predicate::Negation.new(subject_rejector)
+      def ignored_subjects
+        rejectors = config.ignore_expressions.map(&SubjectPrefix.method(:new))
+
+        if rejectors.any?
+          Morpher::Evaluator::Predicate::Boolean::Negation.new(
+            Morpher::Evaluator::Predicate::Boolean::Or.new(rejectors)
+          )
```

Comment:
@dkubb I think till I did my cycle on morpher, I'll not touch this. I'll bascially "push this flag down" to the morpher lib.